server/blob/blob-joining/src/main/java/org/apache/james/blob/joining/JoiningBlobStore.java

```diff
@@ -94,13 +94,10 @@ public InputStream read(BlobId blobId) {
     }
 
     private Optional<InputStream> readFromPrimary(BlobId blobId) throws IOException {
-        InputStream inputStream = primaryBlobStore.read(blobId);
-        byte[] bytes = IOUtils.toByteArray(inputStream);
-        if (hasContent(bytes)) {
-            return Optional.of(new ByteArrayInputStream(bytes));
-        }
-
-        return Optional.empty();
+        return Optional.ofNullable(primaryBlobStore.read(blobId))
+            .map(Throwing.<InputStream, byte[]>function(IOUtils::toByteArray).sneakyThrow())
```

Comment:
if we transform the inputStream in byte array, we loose all the benefits of the InputStream, no?

Reply:
Yeah, but If I don't transform it to byte arrays, there is no way to determine whether that inpustream is empty or not.